### PR TITLE
feat(Authoring): Add branch route with basic params

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- Added "Branch off this step" option in the add step button menu
- Created "Create Branch" route
- Create Branch route allows choosing
  - Number of Branch Paths
  - How to Choose Among Paths
  - Choose Step
  - Choose Item

Note: The individual branch path parameters and the merge step parameter will be added later

## Test
- Make sure the "Branch off this step" button brings you to the "Create Branch" route
- Make sure "Number of Branch Paths" is only valid if it is 2 or more
- If "Score" or "Choice Chosen" are selected, make sure the "Choose Step" and "Choose Item" drop downs show up
- If "Score" is selected, make sure you can only select steps and components that can be scored
- If "Choice Chosen" is selected, make sure you can only select steps and components that are Multiple Choice
- If "Score" or "Choice Chosen" are selected and there is only one component that can be selected, automatically select it
- If you switch from "Score" to "Choice Chosen", and the previously selected component was not a Multiple Choice, unselect it
- When you click the "Submit" button, it should alert the parameters for testing purposes

#1745

### CODE REVIEW

File: src/assets/wise5/authoringTool/add-branch/add-branch.component.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    {
+      value: 'random',
+      text: $localize`Random`
+    },
+    {
+      value: 'tag',
+      text: $localize`Tag`
+    }
+  ];
+  protected componentIdToSelectable: any = {};
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-branch/add-branch.component.ts]: Give more specific type (string => boolean)?

File: src/assets/wise5/authoringTool/add-branch/add-branch.component.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  protected components: any[];
+  protected formGroup: FormGroup = this.fb.group({
+    pathCount: new FormControl('', [Validators.required]),
+    criteria: new FormControl('', [Validators.required])
+  });
+  protected nodeIds: string[];
+  protected nodeIdToSelectable: any = {};
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-branch/add-branch.component.ts]: Give more specific type (string => boolean)?

File: src/assets/wise5/authoringTool/add-branch/add-branch.component.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
+import { Component } from '@angular/core';
+import { TeacherProjectService } from '../../services/teacherProjectService';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+
+@Component({
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-branch/add-branch.component.ts:17]: Should AddBranchComponent be standalone?

File: src/assets/wise5/authoringTool/add-branch/add-branch.component.ts [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    const criteria = this.getCriteria();
+    if (criteria === this.SCORE) {
+      return this.projectService.componentHasWork(component);
+    } else if (criteria === this.CHOICE_CHOSEN) {
+      return component.type === 'MultipleChoice';
+    }
+    return true;
+  }
+
+  private autoSelectComponentIdIfPossible(): void {
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-branch/add-branch.component.ts]: Should ```autoSelectComponentIdIfPossible``` be ```tryAutoSelectComponendId``` to be consistent with other try* function names?

**Review by hirokiterashima — APPROVED** [review]
LGTM. I added some comments inline to consider before merge.

Also, should we remove the ```first``` Input() variable in AddStepButtonComponent since that's no longer being used?

**Comment by geoffreykwan** [comment]
Ok, I've made the changes.

**Pull request merged**